File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_PACHECO_BRYAN_ORLANDO/POO/Ejer3(py y java)/Producto.java
public class Producto {
    // a)
    private String nombre;
    private double precio;
    private int stock;

    // a) 
    public Producto(String nombre,double precio,int stock){
        this.nombre=nombre;
        this.precio=precio;
        this.stock=stock;
    }

    // b) 
    public void vender(int cantidad){
        if(cantidad<=stock){
            stock-=cantidad;
        } else {
            System.out.println("Stock insuficiente");
        }
    }

    // c)
    public void reabastecer(int cantidad){
        stock+=cantidad;
    }

    // d) 
    public static void main(String[] args){
        Producto prod=new Producto("Piñas",3.5,30); // d)
        System.out.println("actualmente tienes "+prod.nombre+" "+prod.stock);
        prod.vender(10); // b)
        System.out.println("vendiendoo X "+prod.nombre+". . . ");
        System.out.println("Stock ahora: "+prod.stock);
        prod.reabastecer(20); // c)
        System.out.println("reabasteciendo....");
        System.out.println("Stock actual: "+prod.stock);
    }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_PACHECO_BRYAN_ORLANDO/POO/Ejer3(py y java)/Producto.py
class Producto:
    # a) 
    def __init__(self, nombre, precio, stock):
        self.nombre=nombre
        self.precio=precio
        self.stock=stock

    # b) 
    def vender(self, cantidad):
        if cantidad<=self.stock:
            self.stock-=cantidad
        else:
            print("Stock insuficiente")

    # c) 
    def reabastecer(self, cantidad):
        self.stock+=cantidad


prod=Producto("Manzanas", 2.5, 50)
print("actualmente tienes",prod.nombre," ",prod.stock)
prod.vender(10)  
print("vendiendo X ",prod.nombre,". . . .")
print("Stock ahora:", prod.stock)
prod.reabastecer(20)
print("reabasteciendo....")
print("Stock actual:", prod.stock)
